### PR TITLE
SAMZA-1874: Refactor SamzaContainer and TaskInstance unit tests to make shared context changes easier

### PR DESCRIPTION
**Pull request by cameronlee314**

### CODE REVIEW

**Comment by cameronlee314** [comment]
@vjagadish1989 @prateekm can you please review this?

File: samza-core/src/test/scala/org/apache/samza/container/TestSamzaContainer.scala [hunk 1]
```diff
@@ -20,36 +20,160 @@
...
+  @Mock
+  private var metrics: SamzaContainerMetrics = _
+  @Mock
+  private var samzaContainerListener: SamzaContainerListener = _
+
+  private var samzaContainer: SamzaContainer = _
+
+  @Before
+  def setup(): Unit = {
+    MockitoAnnotations.initMocks(this)
```
Comment (prateekm) [inline samza-core/src/test/scala/org/apache/samza/container/TestSamzaContainer.scala]: Nice, thanks for cleaning this up.

File: samza-core/src/test/scala/org/apache/samza/container/TestTaskInstance.scala [hunk 2]
```diff
@@ -20,429 +20,211 @@
...
 
-  /*
-   * Helper method used to retrieve the value of a counter from a group.
-   */
-  private def getCount(
-                        group: ConcurrentHashMap[String, Metric],
-                        name: String): Long = {
-    group.get("exception-ignored-" + name.toLowerCase).asInstanceOf[Counter].getCount
+  @Test
+  def testBasicProcess() {
```
Comment (prateekm) [inline samza-core/src/test/scala/org/apache/samza/container/TestTaskInstance.scala]: Minor: Maybe testProcess()?

File: samza-core/src/test/scala/org/apache/samza/container/TestTaskInstanceExceptionHandler.scala [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+import org.apache.samza.metrics.{Counter, MetricsHelper}
+import org.junit.{Before, Test}
+import org.mockito.Mockito._
+import org.mockito.{Mock, MockitoAnnotations}
+import org.scalatest.junit.AssertionsForJUnit
+import org.scalatest.mockito.MockitoSugar
+
+class TestTaskInstanceExceptionHandler extends AssertionsForJUnit with MockitoSugar {
+  @Mock
+  private var metrics: MetricsHelper = _
```
[the remark this answers is not in the document]
Comment (cameronlee314) [reply samza-core/src/test/scala/org/apache/samza/container/TestTaskInstanceExceptionHandler.scala]: The IDE reports `= null` as a "warning" and suggests using `= _`. However, `= null` is more consistent with the current code, so I can change to that.

File: samza-core/src/test/scala/org/apache/samza/system/chooser/MockSystemAdmin.scala [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.samza.system.chooser
+
+import org.apache.samza.system.{SystemAdmin, SystemStreamPartition}
+
+class MockSystemAdmin extends SystemAdmin {
```
[the remark this answers is not in the document]
Comment (cameronlee314) [reply samza-core/src/test/scala/org/apache/samza/system/chooser/MockSystemAdmin.scala]: It was embedded as a separate class in TestTaskInstance.scala, but that test no longer needs it, so I moved it to the only package it was used in.

**Comment by cameronlee314** [comment]
I accidentally messed up this branch with some accidental rebasing and merging. I'm going to close this one and create a new PR with the proper commits just to make sure everything gets reset properly.

**Comment by cameronlee314** [comment]
This is replaced by https://github.com/apache/samza/pull/646

**Pull request closed without merging**